mgmtsystem/azure.py

```diff
@@ -0,0 +1,308 @@
+# coding: utf-8
+"""Backend management system classes
+
+Used to communicate with providers without using CFME facilities
+"""
+import winrm
+import json
+from cStringIO import StringIO
+from contextlib import contextmanager
+from datetime import datetime
+from textwrap import dedent
+
+from lxml import etree
+from wait_for import wait_for
+
+from base import MgmtSystemAPIBase
+
+
+class AzureSystem(MgmtSystemAPIBase):
+    """This class is used to connect to Microsoft Azure Portal via PowerShell AzureRM Module
+    """
+    STATE_RUNNING = "VM running"
+    STATE_STOPPED = "VM deallocated"
+    STATE_STARTING = "VM starting"
+    STATE_PAUSED = "Paused"
+    STATES_STEADY = {STATE_RUNNING, STATE_PAUSED}
+    STATES_STEADY.update(STATE_STOPPED)
```

Comment:
Hehe, actually you don't need this, I used it in SCVMM because there were multiple state string possible for one of the states we recognize and set + set does not work. Feel free to put it in like this:
```python
STATES_STEADY = {STATE_RUNNING, STATE_PAUSED, STATE_STOPPED}
```